src/app/ui/editor/moving_pixels_state.cpp

```diff
@@ -441,7 +441,11 @@ void MovingPixelsState::onCommitMouseMove(Editor* editor,
   KeyAction action = m_editor->getCustomizationDelegate()
     ->getPressedKeyAction(keyContext);
 
-  if (int(action & KeyAction::SnapToGrid))
+  bool snapKeyAction = bool(action & KeyAction::SnapToGrid);
+  if (Preferences::instance().document(editor->document()).grid.snap())
+    snapKeyAction = !snapKeyAction;
+
+  if (snapKeyAction)
```

Comment:
You can use `Editor::docPref()`:
```suggestion
  bool snapKeyAction = bool(action & KeyAction::SnapToGrid);
  if (editor->docPref().grid.snap())
    snapKeyAction = !snapKeyAction;

  if (snapKeyAction)
```
As a minor detail, it looks like the logic is inverted, for example I'd have done something like:
```c++
  bool snapToGrid  = editor->docPref().grid.snap(); // Snap to grid depending on the user preference
  if (bool(action & KeyAction::SnapToGrid))
    snapToGrid = !snapToGrid; // Pressing the key inverts the action

  if (snapToGrid)
```
What do you think?

Reply:
I know, what I said is that the logic is a little confusing to understand (why the Preference inverts the action, when the Alt key is the one that inverts the snap on/off). Both solutions should do the same, but what I think is that it makes more sense to use the preference "as the default", and the Alt modifier as just that, a modifier of the default (inverts the snap on/off depending on the preference).